packages/contracts/test/plugin/plugin-setup-processor.ts

```diff
@@ -697,13 +817,55 @@ describe('Plugin Setup Processor', function () {
       );
     });
 
-    describe('PrepareUpdate', function () {
+    describe('isValidUpdate', function () {
+      it('accepts major updates', async () => {
+        expect(await psp.isValidUpdate([1, 0, 0], [2, 0, 0])).to.equal(true);
+      });
+
+      it('accepts minor updates', async () => {
+        expect(await psp.isValidUpdate([1, 0, 0], [1, 1, 0])).to.equal(true);
+      });
+
+      it('accepts patch updates', async () => {
+        expect(await psp.isValidUpdate([1, 0, 0], [1, 0, 1])).to.equal(true);
+      });
+
+      it('denies update to the same version upgrades', async () => {
```

Comment:
Similar to my previous comment about having a shared bump function,
most of this test can be done on the repo side.

Reply:
Superseded because @mathewmeconry suggested splitting it into two functions.